src/cli.c

```diff
@@ -387,40 +390,106 @@ static int on_generate_resumption_token(quicly_generate_resumption_token_t *self
 
 static quicly_generate_resumption_token_t generate_resumption_token = {&on_generate_resumption_token};
 
-static int send_one(int fd, quicly_datagram_t *p)
+#ifdef __linux__
+
+#ifndef UDP_SEGMENT
+#define UDP_SEGMENT 103
+#endif
+
+static void do_send_gso(int fd, quicly_datagram_t **packets, size_t num_packets, quicly_packet_allocator_t *pa)
 {
+    struct msghdr mess = {};
+    struct iovec vecs[num_packets];
+    union {
+        struct cmsghdr hdr;
+        char buf[CMSG_SPACE(sizeof(uint16_t))];
+    } cmsg = {};
+
+    mess.msg_name = &packets[0]->dest.sa; /* TODO quicly might start sending packets going to different paths? */
```

Comment:
How would you do this? GSO doesn't allow multiple addresses for a GSO burst, right?

Reply:
At the moment, send paths of all the packets being emitted by each call to `quicly_send` is known to be the same. This is because quicly does not (yet) support migration.

Once quicly adds support for migration, `quicly_send` might start returning packets to be sent through more than one path. When that happens, applications using quicly (including src/cli.c) would be required to check the destination address of the packets being provided, group them by paths, and send them separately.

Though I tend to think that we might prefer slightly tweaking the quicly API so that quicly would return packets grouped by paths. That would be a breaking change in terms of API, but would be simpler.